source/concurrent/SynchronizedQueue.ooc

```diff
@@ -63,8 +63,8 @@ BlockedQueue: class <T> extends SynchronizedQueue<T> {
 	}
 	cancel: func { this _waitLock with(|| this _canceled = true) . wakeAll() }
 	wait: func -> T {
-		result: T = null
 		this _waitLock lockWhen(func -> Bool { !this empty || this _canceled })
+		result: T = null
```

Comment:
Related to C code it generates.
This line was previously outside the critical section:
```C
Memory__memcpy(result, &(__SynchronizedQueue_unreferencable4), ((types__Class*)types__Pointer_class())->size);
```
and `types__Pointer_class()` modifies static data.